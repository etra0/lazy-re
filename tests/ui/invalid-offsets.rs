use lazy_re::lazy_re;

#[repr(C, packed)]
#[lazy_re]
struct Foo {
    #[lazy_re(offset = 0x20)]
    a: usize,

    b: usize,
    c: usize,
    #[lazy_re(offset = 0x28)]
    d: usize,
}

fn main() {}
