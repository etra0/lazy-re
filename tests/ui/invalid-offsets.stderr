error[E0080]: evaluation of constant value failed
 --> tests/ui/invalid-offsets.rs:4:1
  |
4 | #[lazy_re]
  | ^^^^^^^^^^ attempt to compute `40_usize - 56_usize`, which would overflow
  |
  = note: this error originates in the attribute macro `lazy_re` (in Nightly builds, run with -Z macro-backtrace for more info)
